include/fiction/algorithms/simulation/sidb/quickexact.hpp

```diff
@@ -0,0 +1,477 @@
+                for (const auto& cell : detected_negative_sidbs)
+                {
+                    charge_lyt_copy.adding_sidb_to_layout(cell, -1);
+                }
+                sim_result.charge_distributions.push_back(charge_lyt_copy);
+            }
+            charge_lyt_new.increase_charge_index_of_sub_layout_by_one(
+                false, false, true,
+                true);  // "false" allows that the charge state of the dependent cell is
+                        // automatically changed based on the new charge distribution.
+        }
+
+        if (charge_lyt_new.is_physically_valid())
+        {
+            charge_distribution_surface<Lyt> charge_lyt_copy{charge_lyt_new};
+            for (const auto& cell : detected_negative_sidbs)
+            {
+                charge_lyt_copy.adding_sidb_to_layout(cell, -1);
+            }
+            sim_result.charge_distributions.push_back(charge_lyt_copy);
+        }
+
+        for (const auto& cell : detected_negative_sidbs)
+        {
+            layout.assign_cell_type(cell, Lyt::cell_type::NORMAL);
+        }
+    }
+    /**
+     * This function is required for the initialization.
+     */
+    void initialize_charge_layout()
+    {
+        // defects are initialized.
+        for (const auto& [cell, defect] : parameter.defects)
+        {
+            if (defect.epsilon_r == 0 && defect.lambda_tf == 0)
+            {
+                charge_lyt.assign_defect(cell,
+                                         sidb_defect{defect.type, defect.charge, charge_lyt.get_phys_params().epsilon_r,
+                                                     charge_lyt.get_phys_params().lambda_tf});
+            }
+            else if (defect.epsilon_r == 0 && defect.lambda_tf != 0)
+            {
+                charge_lyt.assign_defect(
+                    cell, sidb_defect{defect.type, defect.charge, charge_lyt.get_phys_params().epsilon_r});
+            }
+            else if (defect.epsilon_r != 0 && defect.lambda_tf == 0)
+            {
+                charge_lyt.assign_defect(cell, sidb_defect{defect.type, defect.charge, defect.epsilon_r,
+                                                           charge_lyt.get_phys_params().lambda_tf});
+            }
+            else
+            {
+                charge_lyt.assign_defect(cell, defect);
+            }
+        }
+
+        charge_lyt.set_local_external_potential(parameter.local_external_potential);
+        charge_lyt.set_global_external_potential(parameter.global_potential, false);
+        detected_negative_sidb_indices = charge_lyt.negative_sidb_detection();
+        detected_negative_sidbs.reserve(detected_negative_sidb_indices.size());
+        all_sidbs_in_lyt_without_negative_detected_ones = charge_lyt.get_sidb_order();
+        real_placed_defects                             = charge_lyt.get_defects();
+        number_of_SiDBs                                 = charge_lyt.get_sidb_order().size();
```

Comment:
Why do we need these parameters additionally if they are simply equivalent to these respective statements?